### PR TITLE
feat(java-sdk): Merge java_sdk_reactor and java_sdk_1.1.0 into main branch

### PR DESCRIPTION
**Pull request by seeflood**
**What this PR does**:
- Merge java_sdk_reactor and java_sdk_1.1.0 into main branch
- Split the java sdk into different modules.  Users can choose to use `sdk` or `sdk-reactor` based on their preferences.

image

After these different branches being merged into one, we can then move java sdk into the new repo.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by seeflood** [comment]
@kevinten10  @ccx1024cc @ujjboy Hi guys, I'm trying to merge your great work into the main branch.
Could you please help review this PR ?

**Comment by seeflood** [comment]
Is there any way to prevent your commit histories from being deleted while merging these branches?
I'm afraid that currently the `squash and merge` mode will delete your commit histories

File: sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package spec.sdk.reactor.v1.domain.enhanced;
+
+/**
+ * Metrics Runtimes standard API defined.
+ */
+public interface MetricsRuntimes {
```
Comment (kevinten10) [inline sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java:20]: Is there an API definition for application layer observability?

We used `opentelemetry` to make a version of [TelemetryAPI](https://github.com/reactivegroup/cloud-runtimes-jvm/blob/master/cloud-runtimes-api/src/main/java/group/rxcloud/cloudruntimes/domain/enhanced/TelemetryRuntimes.java), which is used for business code to actively initiate metrics/traces.

File: sdk/java-sdk/sdk-reactor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            <extension>
+                <groupId>kr.motd.maven</groupId>
+                <artifactId>os-maven-plugin</artifactId>
+                <version>1.6.2</version>
+            </extension>
+        </extensions>
+        <plugins>
+            <plugin>
+                <groupId>org.xolstice.maven.plugins</groupId>
+                <artifactId>protobuf-maven-plugin</artifactId>
```
Comment (ujjboy) [inline sdk/java-sdk/sdk-reactor/pom.xml:61]: 这一段插件在这个模块里不需要吧

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/client/reactor/LayottoReactorClientGrpc.java [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.mosn.layotto.v1.client.reactor;
+
+import com.google.common.base.Strings;
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Empty;
+import io.grpc.*;
```
Comment (ujjboy) [inline sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/client/reactor/LayottoReactorClientGrpc.java:21]: 不要 import *

**Review by ujjboy — CHANGES_REQUESTED** [review]
整体 pr 有点大，能否把增加 file 和 重构分为2个 pr

File: sdk/java-sdk/sdk-reactor/pom.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            <version>4.9.0</version>
+        </dependency>
+        <!-- necessary for Java 9+ -->
+        <dependency>
+            <groupId>org.apache.tomcat</groupId>
+            <artifactId>annotations-api</artifactId>
+            <version>6.0.53</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
```
Comment (kevinten10) [inline sdk/java-sdk/sdk-reactor/pom.xml:48]: I will manage the version in a unified manner in the future.

**Review by kevinten10 — COMMENTED** [review]
LGTM.

File: sdk/java-sdk/sdk-reactor/pom.xml (hunk 2 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/pom.xml]: fixed

Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/pom.xml]: fixed

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/client/reactor/LayottoReactorClientGrpc.java (hunk 3 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/client/reactor/LayottoReactorClientGrpc.java]: fixed

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Layotto's default IP for gRPC communication.
+     */
+    private static final String DEFAULT_SIDECAR_IP = "127.0.0.1";
+
+    /**
+     * Layotto's default gRPC port.
+     */
+    private static final Integer DEFAULT_GRPC_PORT = 50001;
```
[the remark this answers is not in the document]
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java]: fixed

File: sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java (hunk 1 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java:20]: Currently no,but of course we can discuss whether and how to add it (in a new issue ?)

**Comment by seeflood** [comment]
> 整体 pr 有点大，能否把增加 file 和 重构分为2个 pr

https://github.com/mosn/layotto/pull/356/files 
@ccx1024cc submitted a PR for file API . I will merge his PR to make this PR simpler

File: sdk/java-sdk/sdk-reactor/pom.xml (hunk 4 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/pom.xml:48]: Ok. So I don't have to fix it now :>

Comment (kevinten10) [reply sdk/java-sdk/sdk-reactor/pom.xml:48]: yes

File: sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java (hunk 1 above)
Comment (kevinten10) [reply sdk/java-sdk/sdk-reactor/src/main/java/spec/sdk/reactor/v1/domain/enhanced/MetricsRuntimes.java:20]: yes

File: sdk/java-sdk/sdk-reactor/pom.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    <build>
+        <extensions>
+            <extension>
+                <groupId>kr.motd.maven</groupId>
+                <artifactId>os-maven-plugin</artifactId>
+                <version>1.6.2</version>
+            </extension>
+        </extensions>
```
Comment (ujjboy) [inline sdk/java-sdk/sdk-reactor/pom.xml:57]: this extension is no need, too.

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/utils/GrpcWrapper.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+ * limitations under the License.
+ */
+package io.mosn.layotto.v1.utils;
+
+import io.grpc.*;
+import io.mosn.layotto.v1.config.Property;
+import reactor.util.context.Context;
+import spec.proto.runtime.v1.RuntimeGrpc;
+
+import java.util.logging.Logger;
```
Comment (ujjboy) [inline sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/utils/GrpcWrapper.java:22]: log facade change to slf4j.

**Review by ujjboy — CHANGES_REQUESTED** [review]

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/utils/GrpcWrapper.java (hunk 7 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/utils/GrpcWrapper.java]: fixed

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            "LAYOTTO_GRPC_PORT",
+            DEFAULT_GRPC_PORT);
+
+    /**
+     * Determines if Layotto client will use gRPC to talk to Layotto's sidecar.
+     */
+    public static final Property<LayottoApiProtocol> API_PROTOCOL = new GenericProperty<>(
+            "layotto.api.protocol",
+            "LAYOTTO_API_PROTOCOL",
+            DEFAULT_API_PROTOCOL,
```
Comment (ujjboy) [inline sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: 这部分常量跟其它模块看看是否一致，或者是否匹配

**Review by ujjboy — APPROVED** [review]
LGTM

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java (hunk 8 above)
Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: 好的 后续挪到新仓库后提新PR修改

**Review by zhenjunMa — APPROVED** [review]
LGTM

File: sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java (hunk 8 above)
Comment (kevinten10) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: 后续两种模式的client，是否会合并共同的代码部分呀

Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: @kevinten10 我提个sdk-common模块，供两种模式依赖，这样如何? 或者有啥更好的办法

Comment (kevinten10) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: OK啊，不过sdk-common会不会比较泛指，使用sdk-infrastructure，sdk-domain这些命名会不会更加清晰呀

Comment (seeflood) [reply sdk/java-sdk/sdk-reactor/src/main/java/io/mosn/layotto/v1/config/Properties.java:70]: 好 我改好发你们看看

**Pull request merged**